frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,19 @@
+import { AppRoute, DataStatus } from '~/libs/enums/enums.js';
+import { useAppSelector, useEffect, useNavigate } from '~/libs/hooks/hooks.js';
+
+type Properties = {
+  children: React.ReactNode;
+};
+const ProtectedRoute: React.FC<Properties> = ({ children }: Properties) => {
+  const { auth } = useAppSelector((auth) => auth);
```

Comment:
```suggestion
  const { user } = useAppSelector(({ auth }) => ({ user: auth.user }));
```

Reply:
Thanks a lot. I am really exhausted. But i want to show next  - user slice  const initialState: State = {
  users: [],
  dataStatus: DataStatus.IDLE,
};      type of users type UserGetAllItemResponseDto = {
  id: number;
  email: string;
  firstName: string;
  lastName: string;
};   so your suggest  useAppSelector(({ auth }) => ({ user: auth.user }));    give me an error. I can't  get these dates. I have only {auth, users} where auth= euth.dataSate{idle, pending, fullfilled and rejected} and users = [ { id: number;
  email: string;
  firstName: string;
  lastName: string;},{...},{..}].  And now   there is only  Property 'auth.user' does not exist on type 'State'.

Reply:
I really amazed because only master could change structure repository without documentation. this morning's meet i have asked should i add something or so on. I tooke answer - only responce repo and no else. So i thanks a lot  but how to gess everything like this one it's mistary ....